app/backend/hiStoryBackend/api/models.py

```diff
@@ -123,3 +127,50 @@ def send_password_reset_email(self):
             return sent_email_count != 0
         except SMTPException:
             return False
+
+
+# hiStory Models
+
+
+class BaseModel(models.Model):
+    created = models.DateTimeField(auto_now_add=True)
+    updated = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        abstract = True
+
+
+class MemoryPost(BaseModel):
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    title = models.CharField(max_length=255)
+    story = JSONField()
+    time = JSONField()
+    location = JSONField()
+
+
+def upload_memory_media_to(instance, filename):
+    """
+    Returns a custom name for the file stored in 'file' field of Memory Media.
+    :param instance: MemoryMedia object
+    :param filename: Name of the file stored in 'file' field
+    :return: A string in the format <memory_post_id>_<year>-<month>-<day>_<hour>_<minute>_<second>_<original_file_name>
+    """
```

Comment:
:+1: 
Files will be saved one at a time. I will add "microsecond" part to the file name. I think, this will ensure that the files names will be different even for the same files in a MemoryPost.

Reply:
Why don't we also store in which order the files appear in a memory post in the file name? Yeah, your fix might work but I think this solves the problem from the root. It would be as in
:return: A string in the format <memory_post_id>_<order>_<year>-<month>-<day>_<hour>_<minute>_<second>_<original_file_name>